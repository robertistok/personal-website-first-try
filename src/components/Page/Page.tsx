import React, { useRef, useEffect } from "react";
import styles from "./Page.module.scss";

interface PageProps {
  title?: string;
}

const Page: React.FunctionComponent<PageProps> = ({ title, children }) => {
  const pageRef = useRef(null);

  useEffect(() => {
    pageRef.current.scrollIntoView();
  });

  return (
    <div ref={pageRef} className={styles["page"]}>
      <div className={styles["page__inner"]}>
        {title && <h1 className={styles["page__title"]}>{title}</h1>}
        <div className={styles["page__body"]}>{children}</div>
      </div>
    </div>
  );
};

export default Page;
