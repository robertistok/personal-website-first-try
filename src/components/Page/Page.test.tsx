import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import Page from "./Page";

describe("<Page />", () => {
  const props = {
    children: () => <span>test</span>,
    title: "test"
  };

  it("renders", () => {
    shallow(<Page {...props} />);
  });

  it("renders and matches snapshot", () => {
    const wrapper = shallow(<Page {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
