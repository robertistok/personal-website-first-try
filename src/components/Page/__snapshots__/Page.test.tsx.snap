// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> renders and matches snapshot 1`] = `
<div
  className="page"
>
  <div
    className="page__inner"
  >
    <h1
      className="page__title"
    >
      test
    </h1>
    <div
      className="page__body"
    >
      <Component />
    </div>
  </div>
</div>
`;
